sycl/include/sycl/builtins.hpp

```diff
@@ -170,6 +187,100 @@ inline __SYCL_ALWAYS_INLINE
 
 #undef __SYCL_MATH_FUNCTION_2_OVERLOAD_IMPL
 
+#define __SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD(NAME)                      \
+  template <typename T, size_t N>                                              \
+  inline __SYCL_ALWAYS_INLINE                                                  \
+      std::enable_if_t<detail::is_sgenfloat<T>::value, marray<T, N>>           \
+      NAME(marray<T, N> x, T y) __NOEXC {                                      \
+    marray<T, N> res;                                                          \
+    sycl::vec<T, 2> y_vec{y, y};                                               \
+    for (size_t i = 0; i < N / 2; i++) {                                       \
+      auto partial_res = __sycl_std::__invoke_##NAME<vec<T, 2>>(               \
+          detail::to_vec2(x, i * 2), y_vec);                                   \
+      std::memcpy(&res[i * 2], &partial_res, sizeof(vec<T, 2>));               \
+    }                                                                          \
+    if (N % 2) {                                                               \
+      res[N - 1] = __sycl_std::__invoke_##NAME<T>(x[N - 1], y_vec[0]);         \
+    }                                                                          \
+    return res;                                                                \
+  }
+
+__SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD(fmax)
+    __SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD(fmin)
+
+#undef __SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD
```

Comment:
What is this `#undef` for? Can we move it closer to macro use to make `#define`..`#undef` block smaller?

Reply:
I guess the intention for already implemented funcs above and below to not expose these macros in other parts of SYCL code and in user's code. However, their names too unique and shouldn't make any problems.

> Can we move it closer to macro use to make #define..#undef block smaller?

Currently this block as small as possible

Reply:
Hm, it looks like some typo. Removed it because we don't need it: https://github.com/intel/llvm/pull/8631/commits/0a4f0c6cfcd8080580cb6a1d858d82f02c991af0

Reply:
I guess I found where it should be